packages/typescript/lib/node/proxyLanguageService.ts

```diff
@@ -936,14 +936,29 @@ function provideInlayHints(language: Language<string>, provideInlayHints: ts.Lan
 		if (serviceScript) {
 			let start: number | undefined;
 			let end: number | undefined;
-			const map = language.maps.get(serviceScript.code, targetScript);
+			const map = language.maps.get(serviceScript.code, sourceScript);
```

Comment:
First of all maps were taken for the wrong script. It should be a source script to have correct mappings from the template to generated code